### ISSUE TITLE
Grouping Questions

### ISSUE THREAD

**Issue by rapi-castillo** [SwimmyApp/about#3]
## Chat Logs

[11:16]  
Also, is uploading headshots of ourselves something we do now? :wink:

ben-pr-p [11:20 AM] 
i think we gotta be careful with forcing a voting mechanism on people – like if we choose majority rule we mess with groups that do consensus already and vice versa. we should have lots of different voting plug ins – i tried to implement that in DemocracyOS a while back and it’s do able

[11:22]  
also, “forking” groups (copying all group data and sending an invite to all current group members) should be possible and is a really good work around in the event of broken governance mechanisms

rapi [11:33 AM] 
^ really good point

[11:37]  
And yes! I think forking groups is a great idea, kinda like a low-stress mechanism used in gits too

ben-pr-p [11:45 AM] 
oh last thing, if we wanna make voting a well supported process in the app, we get a bunch of functionality that’s useful for activist groups that i had in the back of my mind like a “twitter quora”, where tweets can get proposed and automatically sent out if approved by `x` people and rejected by none

[11:45]  
very useful for reducing stress on the main tweeters

[11:45]  
obviously not MVP though

mattmallon [11:46 AM] 
I could also see how forking a group and automatically inviting everyone could cause a lot of drama and confusion. Lots to think about around this issue, but I'd still like to focus as much as we can on the bread and butter of this app... What is the bare minimum a typical user would want to do on a typical day? If the core of the app's functionality is still not prototyped and agreed upon, a pivot to that core functionality could throw off all of our other plans and result in wasted effort. Not meaning to hinder discussion or anything here, just trying to keep us on track! (edited)

rapi [11:49 AM] 
totally valid @mattmallon ! It's good to keep this at the back of the head though, this brings up the question of ownership of the groups, imo. Because the amendment focuses on how a group makes decisions and the capacities of an owner/creator. so if the owner has certain "powers", but members can "fork" it, I feel the goal is a check and balance on the powers within a group. (edited)

[11:51]  
so yeah, just an open question, but not blocking i think (edited)

mattmallon [11:51 AM] 
I see, so it's less about the exact mechanism for kicking/banning and more about group ownership and decision making?

rapi [11:51 AM] 
yeah,

[11:51]  
group ownership mainly

[11:52]  
because when you decentralize group ownership, and recognize that the creator of a group is just that, a creator, it opens up a new dynamic. So the kicking mechanism is more of an example of the rethinking

[11:52]  
I think a good example is slack, so a creator has an inherent ownership of a group,

[11:53]  
but not really hardcoded power in it. So when others will rename it or archive a group that totally affects dynamics

[11:55]  
on the other side, the ownership of it as an administrator is the current status quo

[11:55]  
so yeah, i think it's not an issue, it's the same creator_id fk, imo

[11:55]  
but the mechanisms available is good to think about tho not dwell into at the moment :smile:

[11:55]  
will revert back the amendment and just put it in as a comment! (edited)

ben-pr-p [11:59 AM] 
i think long term forking is a good feature for handling the left’s sectarianism / ideological differences – if group A is a coalition of abc, and a doesn’t want to endorse c’s most recent action, a can fork A and create B, which includes ab while A is now bc, and both A and B are still part of ZZ, a coalition of the whole alphabet in Pennsylvania, for example

[11:59]  
the logic there may or may not be followable

joepbreslin [12:15 PM] 
Definitely worth diving into in some detail, this mechanic is going to be one of the key sticking points. Initially I think best bet is to keep it super lightweight, erring toward giving power to the community over the creator.

mattmallon [12:37 PM] 
Yeah @joepbreslin , I just spent forever typing up some ideas to post here, but it’s too long and complicated for anyone to want to read! Ha! Good points @rapi-progcode and @ben-pr-p , you’ve really got me thinking. The way we decide to do things in terms of ownership/voting could change a lot about how the app/groups are structured. My main food for thought is this:

1. Would voting be required for only large-scale changes (name change, etc.), or more regularly, for things like what events the group will put on?

2. If voting is required on a more regular basis, would that inherently favor some types of members over others? Would it favor those who happen to have the luxury of having more free time to check the app for an hour every day to vote on things, vs. members who may not have the luxury of having that much time? Would it exclude folks who have to work multiple jobs or who are taking care of family members, and only have a couple hours a week to get involved? Or even just folks who are more casual and not super-activists? How can we make it more accessible for them?

3. Should we favor in-person decision-making? Could a group admin have the control to change the name, for instance, but have to link to the event at which that was decided in order to make the change?

4. Would more centralized control for the owners be permissible as long as that control was transparent, and with channels for members to speak up and dissent if necessary? (edited)

joepbreslin [12:47 PM] 
No way, @mattmallon, I love reading everything you write! Good thoughts all around, great questions with big implications for the answer. I'd really recommend we hash out through a call at some point as the easiest route if everyone is interested.

[12:48]  
One thing to note, though, is that I'd think @eddowding would have a really good perspective on this from the represent.me app. There may be things already in place to actually help with a lot of this

ben-pr-p [12:48 PM] 
1) i assume so, and configurable (votable) (woah meta)

2) definitely. solutions include delegation and quorum + veto decision making (n / 2 is enough given no one said no)

3) interesting idea that, if it caused more people to actually meet, would make me happy. may exarcebate 2.

4) i think so, and i think forking helps a lot there. it's what i think makes the open source benevolent dictator for life model end up more democratic than majority vote a lot of the time

Pinned by joepbreslin 
Yesterday at 9:02 PM Pinned by joepbreslin
mattmallon [2:02 PM] 
@joepbreslin @euyterho  A meeting this week would definitely be a good idea. It’d be good for out-of-towners like myself to catch up on what transpired at the hackathon. I also have a proposal that I’d like to discuss, that this recent conversation has brought to mind. I’ve been getting tangled up lately on what feature ideas/discussions are pertinent to MVP or post-MVP, and also I’d like to continue to ensure that the features we prioritize are the ones that are actually going to be of use to activists using this app. For instance, I personally think the group ownership stuff we were just discussing is important, but what if we ask the same question to a dozen different activists with real-world experience who all agree that it’s a lower priority issue than something we haven’t even thought about yet? So with that in mind, here is my dream:

1. We clarify our terms with MVP/prototyping stages, document it, and start labeling features accordingly in the trello board. Adding features/changing priority would require a slack vote. (Similar to what Rapi did yesterday with the change in the prototype doc…good idea!)

2. We had been discussing a bit some time ago doing UX research and identifying a list of problems that our app is intended to solve. I’d like to restart that discussion. Could we get a group of activists with experience, either from within ProgCode, outside ProgCode, or potentially through Autumn and Action Groups Network, and have them be a resource for us whenever we have questions about what direction to go in? An activist round table of sorts, who would basically be serving the role of client for us? First step is identifying and reaching out to those who would be willing (a minimum of 3-5 people I would think, but the more perspectives we get, the better…30 self-selected participants is usually the minimum goal in terms of getting relevant statistics). Hopefully we wouldn’t have to bug them more than once every couple weeks.

3. We all agree on a list of initial survey questions to send to the activist round table that would give us the general picture of what their problems are and what solutions they might be interested in, then gather our data.

4. Compile a list of problems our app is solving, in priority order based on the responses, and as a measure, whenever a new feature idea is proposed, it needs to map to one of those problems. That way, we stay on track to what is of most importance to our users.

5. When we’ve gathered a list of features that map to the high priority problems and that would feasibly be part of an MVP, we ask the activists for their feedback on it to make sure they agree that our proposed features would indeed be helpful, and to identify shortcomings we may have missed.

6. When we have our first front-end prorotype ready, send off another survey with a link to the prototype, for them to submit their feedback on the direction we’re headed in.

In general, I would like to identify clients/shareholders for this project, whom we would consider the subject matter experts, and whom would be available for us to ask clarifying questions. Their experience would help us think of advantages and disadvantages that we may not be familiar with. Since our goal here is to serve as many activists as possible rather than just one group, we would need a number of different clients rather than just one. I think that will help us immensely in not getting ourselves tied up in knots over proposed functionality or prioritization, and ensure that our hard work on this will be of maximum utility. If we have disagreements about how to build things, it’ll be easier to defer to the experts and their collective knowledge to decide how to move forward.

I just wanted to put that out there for folks to consider, and hopefully we can discuss it at a meeting soon after we’ve all had time to ponder. As usual, sorry for writing a novel… (Did I mention I was a writer before I was a programmer?!)

rapi [3:21 PM] 
PROJECT SWIMMY's PASSED THE QUALIFYING ROUND!!

[3:22]  
We PRESENT TO THE JUDGES!!

erika [3:52 PM] 
WHOAAA

rapi [3:55 PM] 
We're getting interviewed by Vice News!

erika [4:01 PM] 
whaaaaaaaat

[4:01]  
awesome!

[4:03]  
I still need to catch up the awesome discussion above but I did see the part about needing a meeting and I totally agree! Would love to make sure @tinobedi @rapi-progcode @florida and/or @sean.mac can come so we can discuss a debrief of the amazing hackathon. (edited)

erika [5:38 PM] 
still waiting on Swimmy on the live feed, sitting on the edge of my seat now hahaha

rapi [5:40 PM] 
Swimmy will be fourth after the break!

erika [5:57 PM] 
Swimmy is after this current presentation

[5:57]  
I can't wait :heart:

dave_mahler [6:00 PM] 
On deck!

[6:00]  
Get your popcorn popped

erika [6:07 PM] 
I'M SO FRICKIN PROUD OF YOU ALL

preston [6:07 PM] 
Wow AMAZING!!!

erika [6:07 PM] 
THAT WAS AWESOME!!! (cheering you all on at midnight in Paris in front of my laptop)

mattmallon [6:15 PM] 
Amazed at how much information you guys could convey in like 2 minutes! Superstars! Great job! I would’ve surely still been at the awkward opening joke/pun phase when they called time… Congrats!

tinobedi [8:48 PM] 
@sky_moveon came as well!!

[8:50]  
I had an awesome time and we made great leaps too

joepbreslin [9:07 PM] 
Didn't want this to get lost, @mattmallon, good thoughts all around and agreed, those are some pretty important things to consider.

To your comments, one suggestion is that once the prototype is available we quickly prioritize building a feedback mechanism directly into the experience (could be targeted to specific users or any other categorization).

I also recommend Prog Code being one of the groups to adopt and help stress test this tool from the beginning, as well as being a great connecting point for the users you mention.

As for the decision making, I agree that we should prioritize intelligently, though I could definitely see major value to this up front, especially if we adopt it, as it could help solve one of our biggest challenges, decision making.

As an aside, I've long had an idea about an easier and more effective means to gather focus group type information, it sounds like this could lead to it one way or another.

mattmallon [9:10 PM] 
Great ideas @joepbreslin! You're right, continuously getting feedback after it's live is also huge!

erika [8:03 AM] 
Hey all,
I've been reading Why David Sometimes Wins by Marshall Ganz, he's in many ways the father of organizing (following Saul Alinsky's legacy) with decades of experience in community, labor, and electoral organizing, starting in the civil rights movement and the United Farm Worker's movement. He was the architect behind the most effective training materials for Obama and Bernie.

There are many thought provoking ideas in this book pertaining to Swimmy and Progcode:

Sustaining a creative deliberate process, however, is challenging and requires leadership with a high tolerance for ambiguity. We know that deliberation that is open to "deviant"- that is contrary-perspectives enhances learning, innovation, and the performance of cognitive tasks in general. But because minorities tend to conform to majorities, and person with less authority tend to conform to those with more authority, a group's tendency over time is to lose its diversity. Particular organizational practices are thus required to preserve diverse perspectives. For brainstorming to give way to decision making, deliberative practices that encourage divergent thinking must also allow for convergent thinking. Conflict resolution by negotiation accompanied by voting is thus preferable to decision making by either a fiat or consesus, because negotiation and voting make collective action possible while preserving the differences that are so useful in deliberation. Moreover, if a leadership team strategizes and acts at the same time, as is the case in a rapidly unfolding social movement, managing these two deliberate modalities-divergent and convergent-poses a special challenge.

Clips of the book are available here: https://books.google.se/books?id=wEkr18AW-SkC&lpg=PT25&ots=h18B3Wc3sD&dq=in%20all%20of%20these%20ways%20a%20leadership%20team's%20strategic%20capacity%20marshall%20ganz&hl=fr&pg=PT26#v=onepage&q&f=false

[8:08]  
A second important structural influence on strategic capacity derives from the kind of resources on which the organization relies. For example, organizations that depend on constituency-based, task-generated resources (e.g. dues) must devise strategies to which their constituents respond. By contrast, organizations that rely on outside resources (e.g. grants) can be less responsive to the constituencies that are critical to their strategic success. It is often the case, for example, that reliance on outside resources can discourage learning-in fact, as long as the bills keep getting paid, leaders of such organizations can keeping doing the same things wrong. An organization that generates the most strategic capacity, however, by drawing resources from multiple salient constituencies. This arrangement allows leaders the most room to maneuver while at the same time affording them the benefits of feedback from a diversity of constituencies.

mattmallon [10:47 AM] 
@euyterho erika  Amazing! So pertinent to what we were just discussing yesterday! Would love to save this somewhere so that it doesn't get lost in the stream of slack conversations. And also to gather quotes/ideas from other resources, books, etc. I made a "Research" sub-folder in the Swimmy drive folder, and started a document with your excerpts, as a place that anyone can contribute if they have other resources to recommend: https://docs.google.com/document/d/1KYzqyX6yC2Q2k-eElJ-PK6XvYpKV-lVidwS4MtlWXqk/edit?usp=sharing Definitely would be great to ground this not only in the efforts of activists today, but also in the greater historical context.

erika [10:49 AM] 
Thanks @mattmallon, my work hasn't given me much to do today so I've mostly been getting my mind blown by writing on organizing, so much to learn!

[10:50]  
Just one more thing (and I'll add it to the doc) but if you have 10 minutes (or 5 if you watch vids at 2x speed) check out the start of his lecture on a course on organizing as a form of leadership: https://youtu.be/zpMzRRaDZzU?t=9m56s
